deploy/docs/v2_migration_doc.md

```diff
@@ -82,3 +82,87 @@ the exact steps for migration.
+```
+
+#### Update
+
+Run the command shown below to fetch the latest helm chart:
+
+```bash
+helm repo update
+```
+
+For users who are not already on `v1.3.2` of the helm chart, please upgrade
+to that version first by running the below command:
+
+```bash
+helm upgrade collection sumologic/sumologic --reuse-values --version=1.3.2
+```
+
+### 2. Upgrade Prometheus CRDs
+
+Due to changes in `kube-prometheus-stack` which this chart depends on, one will
+need to run the following commands in order to update Prometheus related CRDs.
+
+```bash
+kubectl apply -f https://raw.githubusercontent.com/prometheus-operator/prometheus-operator/release-0.43/example/prometheus-operator-crd/monitoring.coreos.com_probes.yaml
+kubectl apply -f https://raw.githubusercontent.com/prometheus-operator/prometheus-operator/release-0.43/example/prometheus-operator-crd/monitoring.coreos.com_alertmanagers.yaml
+kubectl apply -f https://raw.githubusercontent.com/prometheus-operator/prometheus-operator/release-0.43/example/prometheus-operator-crd/monitoring.coreos.com_alertmanagerconfigs.yaml
+kubectl apply -f https://raw.githubusercontent.com/prometheus-operator/prometheus-operator/release-0.43/example/prometheus-operator-crd/monitoring.coreos.com_prometheuses.yaml
+kubectl apply -f https://raw.githubusercontent.com/prometheus-operator/prometheus-operator/release-0.43/example/prometheus-operator-crd/monitoring.coreos.com_prometheusrules.yaml
+kubectl apply -f https://raw.githubusercontent.com/prometheus-operator/prometheus-operator/release-0.43/example/prometheus-operator-crd/monitoring.coreos.com_servicemonitors.yaml
+kubectl apply -f https://raw.githubusercontent.com/prometheus-operator/prometheus-operator/release-0.43/example/prometheus-operator-crd/monitoring.coreos.com_podmonitors.yaml
+kubectl apply -f https://raw.githubusercontent.com/prometheus-operator/prometheus-operator/release-0.43/example/prometheus-operator-crd/monitoring.coreos.com_thanosrulers.yaml
+```
+
+### 3. Run upgrade script
+
+For Helm users, the only breaking changes are the renamed config parameters.
+For users who use a `values.yaml` file, we provide a script that users can run
+to convert their existing `values.yaml` file into one that is compatible with the major release.
+
+- Get the existing values for the helm chart and store it as `current_values.yaml`
+  with the below command:
+
+  ```bash
+  helm get values <RELEASE-NAME> > current_values.yaml
+  ```
+
+- Download the upgrade script via:
+
+  ```bash
+  curl -LJO https://raw.githubusercontent.com/SumoLogic/sumologic-kubernetes-collection/main/deploy/helm/sumologic/upgrade-2.0.0.sh
+  ```
+
+- Run the upgrade script on the above file with the below command.
+
+  ```bash
+  chmod +x upgrade-2.0.0.sh && ./upgrade-2.0.0.sh current_values.yaml
+  ```
+
+- At this point, users can then run:
+
+  ```bash
+  helm upgrade collection sumologic/sumologic --version=2.0.0 -f new_values.yaml
```

Comment:
As 2.0.0 release is not available yet. I tested with dev version - 2.0.0-dev.0-133-g6e362ae, so I added dev helm repo:
```helm repo list
NAME         	URL
sumologic    	https://sumologic.github.io/sumologic-kubernetes-collection
sumologic-dev	https://sumologic.github.io/sumologic-kubernetes-collection/dev
```
and I see following error:
```
$ helm upgrade collection sumologic-dev/sumologic --version=2.0.0-dev.0-133-g6e362ae -f new_values.yaml -n sumologic
Error: UPGRADE FAILED: cannot patch "collection-fluent-bit" with kind DaemonSet: DaemonSet.apps "collection-fluent-bit" is invalid: spec.selector: Invalid value: v1.LabelSelector{MatchLabels:map[string]string{"app.kubernetes.io/instance":"collection", "app.kubernetes.io/name":"fluent-bit"}, MatchExpressions:[]v1.LabelSelectorRequirement(nil)}: field is immutable && cannot patch "collection-sumologic-fluentd-events" with kind StatefulSet: StatefulSet.apps "collection-sumologic-fluentd-events" is invalid: spec: Forbidden: updates to statefulset spec for fields other than 'replicas', 'template', and 'updateStrategy' are forbidden && cannot patch "collection-sumologic-fluentd-metrics" with kind StatefulSet: StatefulSet.apps "collection-sumologic-fluentd-metrics" is invalid: spec: Forbidden: updates to statefulset spec for fields other than 'replicas', 'template', and 'updateStrategy' are forbidden && cannot patch "collection-sumologic-fluentd-logs" with kind StatefulSet: StatefulSet.apps "collection-sumologic-fluentd-logs" is invalid: spec: Forbidden: updates to statefulset spec for fields other than 'replicas', 'template', and 'updateStrategy' are forbidden
```

Reply:
So this is most likely connected to labels and/or statefulset's volume claim when `fluentd.persistence` is enabled.

https://github.com/SumoLogic/sumologic-kubernetes-collection/blob/d1ed616c0cb4ae3c46c258eacf48c8eaaac6b22b/deploy%2Fhelm%2Fsumologic%2Ftemplates%2Fstatefulset.yaml#L151-L161

I'll add warning to upgrade 2.0.0 script, make it leave persistence as disabled if user set it as such and add short doc about how to enable it.